### PR TITLE
Use <section> tag rather than <a name=> for terminology.  Use CamelCase for internal name links.  Resolves #455.

### LINKED ISSUE #455: Terminology definitions are confusing respec processor
The term definitions in caliper-spec-v1p2.html use the `name` attribute:
```
<a name="actorDef"></a>
```
This confuses the latest version of w3c\respec and it can't resolve links to `#actorDef`.

We'd prefer it if you could change the them to use the `id` attribute like this:
```
<a id="actorDef"></a>
```
But if that is too hard, we could add a workaround to the IMS version of Respec.

### PR DESCRIPTION
**Pull request by arwhyte**
Resolves #455.

### CODE REVIEW

File: caliper-spec-v1p2.html [hunk 1]
```diff
@@ -114,144 +114,215 @@ <h3>Design Goals and Rationale</h3>
...
+                    connection to the <code>object</code> at a particular moment in time.</p>
+            </section>
 
-            <p><a name="jsonldDef"></a><strong>JSON-LD</strong>: a specification providing a JSON-based data serialization
-                and messaging format, processing algorithms and API for working with <a href="#linkedData">Linked Data</a>.
-                The messages described in this specification are intended to be used in programming environments that
-                support <a href="http://json-ld.org/spec/latest/json-ld/">JSON-LD</a>.</p>
+            <section id="jsonldDef" class='notoc'>
+                <p><strong>JSON-LD</strong>: a specification providing a JSON-based data
+                    serialization and messaging format, processing algorithms and API for working
```
Comment (ssciolla) [inline caliper-spec-v1p2.html]: Maybe change to "processing algorithms and an API"?

File: caliper-spec-v1p2.html [hunk 2]
```diff
@@ -114,144 +114,215 @@ <h3>Design Goals and Rationale</h3>
...
+                    with <a href="#linkedData">Linked Data</a>. The messages described in this
+                    specification are intended to be used in programming environments that
+                    support <a href="http://json-ld.org/spec/latest/json-ld/">JSON-LD</a>.</p>
+            </section>
 
-            <p><a name="iriDef"></a><strong>IRI</strong>: The Internationalized Resource Identifier (IRI) extends the
+            <section id="iriDef" class='notoc'>
+                <p><strong>IRI</strong>: The Internationalized Resource Identifier (IRI) extends the
                 Uniform Resource Identifier (<a href="#uriDef">URI</a>) scheme by using characters drawn from the Universal
```
Comment (ssciolla) [inline caliper-spec-v1p2.html]: The reference to the "Universal character set" could be updated to reference Unicode or its associated ISO standard.

File: caliper-spec-v1p2.html [hunk 3]
```diff
@@ -114,144 +114,215 @@ <h3>Design Goals and Rationale</h3>
...
-                their <a href="#iriDef">IRIs</a>/<a href="#uriDef">URIs</a> in order to promote discovery of new relationships
-                between things.</p>
+            <section id="linkedDataDef" class='notoc'>
+                <p><strong>Linked Data</strong>: A set of design principles first articulated by
+                    Tim Berners-Lee for discovering, connecting, and sharing structured data over
+                    the Web. The principles can be summarized as follows: use
+                    <a href="#iriDef">IRIs</a> /<a href="#uriDef">URIs</a> as names for things;
+                    use HTTP <a href="#iriDef">IRIs</a> /<a href="#uriDef">URIs</a> so that
+                    information about things (e.g., people, objects, concepts) can be retrieved
+                    using a standard format; link out to other relevant things by way of
```
Comment (ssciolla) [inline caliper-spec-v1p2.html:232]: Add "and" before "link out"?

File: caliper-spec-v1p2.html [hunk 4]
```diff
@@ -114,144 +114,215 @@ <h3>Design Goals and Rationale</h3>
...
+                    <a href="#rfc2396">RFC 2396</a> a <a href="#uriDef">URI</a> "can be further
+                    classified as a locator, a name or both."  Both the Uniform Resource Locator
+                    (<a href="#urlDef">URL</a>) and the Uniform Resource Name
+                    (<a href="#urnDef">URN</a>) are considered subspaces of the more general
+                    <a href="#uriDef">URI</a> space.</p>
+            </section>
+
+            <section id="urlDef" class='notoc'>
+                <p><strong>URL</strong>: A Uniform Resource Locator (<a href="#urlDef">URL</a>) is a
+                    type of <a href="#uriDef">URI</a> that provides a reference to resource that
```
Comment (ssciolla) [inline caliper-spec-v1p2.html]: Perhaps change to "that provides a reference to a resource".

File: caliper-spec-v1p2.html [hunk 5]
```diff
@@ -114,144 +114,215 @@ <h3>Design Goals and Rationale</h3>
...
+                    specifies both its location and a means of retrieving a representation of it.
+                    An HTTP <a href="#uriDef">URI</a> is a <a href="#urlDef">URL</a> and using HTTP
```
Comment (ssciolla) [inline caliper-spec-v1p2.html]: Add "," after URL a tag.

File: caliper-spec-v1p2.html [hunk 6]
```diff
@@ -114,144 +114,215 @@ <h3>Design Goals and Rationale</h3>
...
+                    <a href="#iriDef">IRIs</a> /<a href="#uriDef">URIs</a> to identify things is
+                    fundamental to <a href="#linkedData">Linked Data</a>.</p>
+            </section>
+
+            <section id="urnDef" class='notoc'>
+                <p><strong>URN</strong>: A Uniform Resource Name (<a href="#urnDef">URN</a>) is a
+                    type of <a href="#uriDef">URI</a> that provides a persistent identifier for a
+                    resource that is bound to a defined namespace.  Unlike a
+                    <a href="#urlDef">URL</a> a <a href="#urnDef">URN</a> is location-independent
```
Comment (ssciolla) [inline caliper-spec-v1p2.html]: Add "," after URL a tag.

File: caliper-spec-v1p2.html (hunk 1 above)
Comment (arwhyte) [reply caliper-spec-v1p2.html]: Done

File: caliper-spec-v1p2.html (hunk 2 above)
Comment (arwhyte) [reply caliper-spec-v1p2.html]: Can you provide the reference?

File: caliper-spec-v1p2.html (hunk 3 above)
Comment (arwhyte) [reply caliper-spec-v1p2.html:232]: use of semi colon cancels out need to add "and", as in 

```
do 1;
do 2;
do 3
```

yeah?

File: caliper-spec-v1p2.html (hunk 4 above)
Comment (arwhyte) [reply caliper-spec-v1p2.html]: Done

File: caliper-spec-v1p2.html (hunk 6 above)
Comment (arwhyte) [reply caliper-spec-v1p2.html]: Done

**Comment by arwhyte** [comment]
Performed a regex search using the expression `"#[a-z]` across the caliper 1.2 spec doc as well as `/fragments` files.  Fixed bad links.  We could do further work in this area such as standardizing all link references using camel case with capitalization of first letter of each word.  @ssciolla -- what say you?

File: caliper-spec-v1p2.html (hunk 5 above)
Comment (arwhyte) [reply caliper-spec-v1p2.html]: Done

File: caliper-spec-v1p2.html [hunk 7]
```diff
@@ -264,13 +264,13 @@ <h3>Terminology</h3>
 
             <section id="sensorDef" class='notoc'>
                 <p><strong>Sensor</strong>: Software assets deployed within a learning application
-                    that implement the <a href="#sensorAPIDef">Sensor API&trade;</a> for
+                    that implement the <a href="#SensorAPIDef">Sensor API&trade;</a> for
```
Comment (ssciolla) [inline caliper-spec-v1p2.html:267]: The change here broke this link.

File: caliper-spec-v1p2.html [hunk 8]
```diff
@@ -264,13 +264,13 @@ <h3>Terminology</h3>
...
                     marshalling and transmitting Caliper data to a target endpoint.</p>
             </section>
 
             <section id="sensorAPIDef" class='notoc'>
                 <p><strong>Sensor API&trade;</strong>: The standard set of methods and supported
-                    parameters that a <a href="#sensorDef">Sensor</a> implements according to this
+                    parameters that a <a href="#SensorDef">Sensor</a> implements according to this
```
Comment (ssciolla) [inline caliper-spec-v1p2.html:273]: The change here broke this link.

**Comment by arwhyte** [comment]
@ssciolla--two additional changes I made:

1. endpoint link capitalized
2. sensor link capitalized

File: caliper-spec-v1p2.html (hunk 3 above)
Comment (ssciolla) [reply caliper-spec-v1p2.html:232]: This is probably subjective, but to me it felt like a list. Maybe the semicolons can go away and be replaced by commas and an "and" where I suggested it. If you like it the way it is, that's fine, too.

File: caliper-spec-v1p2.html (hunk 2 above)
Comment (ssciolla) [reply caliper-spec-v1p2.html]: This likely deserves more research, but the [ISO Standard 10646](https://www.iso.org/standard/69119.html) appears to be specifying a "Universal Coded Character Set" now, not a "Universal character set". This [FAQ page](https://www.unicode.org/faq/unicode_iso.html) from Unicode may also be helpful in explaining the relationship between Unicode and the ISO Standard.

**Comment by ssciolla** [comment]
@arwhyte Making sure you saw my comments about the problems with the sensor-related definition links.

**Comment by arwhyte** [comment]
@ssciolla -- changed Profiles, Events, Entities, Actions, Metrics, Roles, Selectors, Statuses link names to CamelCase as well as fixed numerous naming errors encountered.

**Comment by arwhyte** [comment]
Will change all other link names to CamelCase with leading Capitalized letter to ensure consistency.

File: fragments/events/caliper-event-search.html [hunk 9]
```diff
@@ -56,17 +56,17 @@ <h3>SearchEvent</h3>
     <tr>
         <td>action</td>
         <td><a href=
-           "https://www.imsglobal.org/sites/default/files/caliper/v1p1/caliper-spec-v1p1/caliper-spec-v1p1.html#termDef"
+           "https://www.imsglobal.org/sites/default/files/caliper/v1p1/caliper-spec-v1p1/caliper-spec-v1p1.html#TermDef"
         >Term</a></td>
         <td>The action or predicate that binds the <code>actor</code> or subject to the
-            <code>object</code>. The value range is limited to the <a href="#actions">Searched</a> action only.
+            <code>object</code>. The value range is limited to the <a href="#Actions">Searched</a> action only.
```
Comment (ssciolla) [inline fragments/events/caliper-event-search.html]: `<a/>` here should be `<a href="Action-searched">Searched</>`

File: fragments/events/caliper-event-toollaunch.html [hunk 10]
```diff
@@ -103,12 +103,12 @@ <h3>ToolLaunchEvent</h3>
...
             <a href="#LtiSession">LtiSession</a> |
             <a href=
-               "https://www.imsglobal.org/sites/default/files/caliper/v1p1/caliper-spec-v1p1/caliper-spec-v1p1.html#iriDef"
+               "https://www.imsglobal.org/sites/default/files/caliper/v1p1/caliper-spec-v1p1/caliper-spec-v1p1.html#IriDef"
             >IRI</a></td>
         <td>The Platform's session, constituting part of the tool launch context. The <code>federatedSession</code>
             value MUST be expressed either as an object or as a string corresponding to the federatedSession’s IRI.
         </td>
-        <td>Required when the <code>action</code> value is <a href="#actions">Launched</a>, otherwise optional.
+        <td>Required when the <code>action</code> value is <a href="#Actions">Launched</a>, otherwise optional.
```
Comment (ssciolla) [inline fragments/events/caliper-event-toollaunch.html]: `<a/>` here should be `<a href="Action-launched">Launched</a>`.

File: fragments/events/caliper-event-tooluse.html [hunk 11]
```diff
@@ -33,51 +33,51 @@ <h3>ToolUseEvent</h3>
...
         </td>
         <td>Required</td>
     </tr>
     <tr>
         <td>action</td>
-        <td><a href="#termDef">Term</a></td>
+        <td><a href="#TermDef">Term</a></td>
         <td>The action or predicate that binds the <code>actor</code> or subject to the <code>object</code>. The
-            value range is limited to the <a href="#actions">Used</a> action only.
+            value range is limited to the <a href="#Actions">Used</a> action only.
```
Comment (ssciolla) [inline fragments/events/caliper-event-tooluse.html]: `<a/>` here should be `<a href="Action-used">Used</a>`.

File: caliper-spec-v1p2.html (hunk 7 above)
Comment (ssciolla) [reply caliper-spec-v1p2.html:267]: Fixed.

File: caliper-spec-v1p2.html (hunk 8 above)
Comment (ssciolla) [reply caliper-spec-v1p2.html:273]: Fixed.

File: caliper-spec-v1p2.html [hunk 12]
```diff
@@ -1692,130 +1691,130 @@ <h2>Actions</h2>
 
         <p>Each Caliper <a href="#Event">Event</a> type supports one or more of the actions listed
             below. The <a href="#Event">Event</a> <code>action</code> property string value MUST be
-            set to the appropriate <a href="termDef">Term</a>. Only one action may be specified per
+            set to the appropriate <a href="TermDef">Term</a>. Only one action may be specified per
```
Comment (ssciolla) [inline caliper-spec-v1p2.html]: Link needs a leading `#`.

File: caliper-spec-v1p2.html [hunk 13]
```diff
@@ -1676,14 +1675,14 @@ <h2>System Identifiers</h2>
...
         <h2>Actions</h2>
 
         <p>Caliper includes a vocabulary of actions for describing learning interactions. Each
-            action <a href="#termDef">Term</a> is based on the past-tense form of an English (en-US)
-            verb. An action <a href="#termDef">Term</a> can also indicate a change in a particular
+            action <a href="#TermDef">Term</a> is based on the past-tense form of an English (en-US)
+            verb. An action <a href="#TermDef">Term</a> can also indicate a change in a particular
             characteristic of the <code>object</code> (e.g., resolution, size, speed, volume). Each
-            action <a href="#termDef">Term</a> is mapped to a persistent <a href="iriDef">IRI</a>
+            action <a href="#TermDef">Term</a> is mapped to a persistent <a href="IriDef">IRI</a>
```
Comment (ssciolla) [inline caliper-spec-v1p2.html]: Link needs a leading `#`.

Comment (arwhyte) [reply caliper-spec-v1p2.html]: Fixed.

File: caliper-spec-v1p2.html (hunk 12 above)
Comment (arwhyte) [reply caliper-spec-v1p2.html]: Fixed.

File: fragments/events/caliper-event-tooluse.html (hunk 11 above)
Comment (arwhyte) [reply fragments/events/caliper-event-tooluse.html]: Fixed

File: fragments/events/caliper-event-toollaunch.html (hunk 10 above)
Comment (arwhyte) [reply fragments/events/caliper-event-toollaunch.html]: Fixed.

File: fragments/events/caliper-event-search.html (hunk 9 above)
Comment (arwhyte) [reply fragments/events/caliper-event-search.html]: Fixed.

File: fragments/entities/caliper-entity-query.html [hunk 14]
```diff
@@ -30,17 +30,17 @@ <h3>Query</h3>
...
+           "#TermDef"
         >Term</a></td>
         <td>The string value MUST be set to the term <em>Query</em>.</td>
         <td>Required</td>
     </tr>
     <tr>
         <td>creator</td>
         <td><a href=
-           "https://www.imsglobal.org/sites/default/files/caliper/v1p1/caliper-spec-v1p1/caliper-spec-v1p1.html#person"
+           "#person"
```
Comment (ssciolla) [inline fragments/entities/caliper-entity-query.html]: There still appear to be some un-capitalized entities.

**Comment by arwhyte** [comment]
@ssiolla--I just pushed a couple of more commits that fix another set of bad links. Do another couple of regex passes and let me know what you find.

**Comment by ssciolla** [comment]
@arwhyte  I ran the `href="[a-z]` and `id="[a-z]` regular expressions on the `caliper-spec-v1p2.html` and on the `fragments` directory. Nothing turned up in `fragments`, but I found a few things (listed below) in the spec document. I also tried adding `[\s]+` in front of the quotation mark in both, but I didn't turn up anything (it looks like you took care of those irregularities already).

1) `#rdf`, `#rfc4122`, `#rfc6750`, `#rfc7231`, and `#rfc7807` are not in CamelCase and do not link anywhere currently.

2) There are a good number of sections and subsections that do not use CamelCase, such as `#serialization`, `#terminology`, `#documentSet`, `#httpRequest`, `#information_model`, etc. Perhaps we should decide if we want these to be part of that naming scheme. If so, the sections and links found by the above regular expressions will need to be updated.

3) `#infoModel` should be `#information_model`, but see 2. 

4) There is a link for an "Appendix G. Minimum Supported String Lengths" (`#minSupportedStringLengths`), but there is no such appendix or id in the document (there was in the v1p1 spec).

**Comment by joshuamcghee** [comment]
Bracken to review PR

**Pull request closed without merging**